app/http.php

```diff
@@ -0,0 +1,313 @@
+                }
+            } catch (\Exception $err) {
+                throw $err;
+            }
+        });
+    }
+}
+
+/**
+ * Create logger instance
+ */
+$providerName = App::getEnv('OPEN_RUNTIMES_PROXY_LOGGING_PROVIDER', '');
+$providerConfig = App::getEnv('OPEN_RUNTIMES_PROXY_LOGGING_CONFIG', '');
+$logger = null;
+
+if (!empty($providerName) && !empty($providerConfig) && Logger::hasProvider($providerName)) {
+    $classname = '\\Utopia\\Logger\\Adapter\\' . \ucfirst($providerName);
+
+    /**
+     * @var Adapter $adapter
+     */
+    $adapter = new $classname($providerConfig);
+    $logger = new Logger($adapter);
+}
+
+function logError(Throwable $error, string $action, Utopia\Route $route = null): void
+{
+    global $logger;
+
+    if ($logger) {
+        $version = App::getEnv('OPEN_RUNTIMES_PROXY_VERSION', 'UNKNOWN');
+
+        $log = new Log();
+        $log->setNamespace('proxy');
+        $log->setServer(\gethostname() ? \gethostname() : 'unknown');
+        $log->setVersion($version);
+        $log->setType(Log::TYPE_ERROR);
+        $log->setMessage($error->getMessage());
+
+        if ($route) {
+            $log->addTag('method', $route->getMethod());
+            $log->addTag('url', $route->getPath());
+        }
+
+        $log->addTag('code', $error->getCode());
+        $log->addTag('verboseType', get_class($error));
+
+        $log->addExtra('file', $error->getFile());
+        $log->addExtra('line', $error->getLine());
+        $log->addExtra('trace', $error->getTraceAsString());
+        $log->addExtra('detailedTrace', $error->getTrace());
+
+        $log->setAction($action);
+
+        $isProduction = App::getEnv('OPEN_RUNTIMES_PROXY_ENV', 'development') === 'production';
+        $log->setEnvironment($isProduction ? Log::ENVIRONMENT_PRODUCTION : Log::ENVIRONMENT_STAGING);
+
+        $responseCode = $logger->addLog($log);
+        Console::info('Proxy log pushed with status code: ' . $responseCode);
+    }
+
+    Console::error('[Error] Type: ' . get_class($error));
+    Console::error('[Error] Message: ' . $error->getMessage());
+    Console::error('[Error] File: ' . $error->getFile());
+    Console::error('[Error] Line: ' . $error->getLine());
+}
+
+Console::success("Waiting for executors to start...");
+
+\sleep(5);
+
+fetchExecutorsState(true);
+
+Console::log("State of executors at startup:");
+
+go(function () use ($executorStates) {
+    $executors = \explode(',', App::getEnv('OPEN_RUNTIMES_PROXY_EXECUTORS', ''));
+
+    foreach ($executors as $executor) {
+        $state = $executorStates->exists($executor) ? $executorStates->get($executor) : null;
+
+        if ($state === null) {
+            Console::warning('Executor ' . $executor . ' has unknown state.');
+        } else {
+            Console::log('Executor ' . $executor . ' is ' . ($state['status'] ?? 'unknown') . '.');
+        }
+    }
+});
+
+Swoole\Event::wait();
+
+$run = function (SwooleRequest $request, SwooleResponse $response) {
```

Comment:
We should use our HTTP framework and try to utilize dependency injections as much as possible. We should avoid using the global scope as much as we can.

Reply:
I don't think we can use our HTTP framework here. Our framework does not support wildcard endpoint, as far as I am concerned (action that would run for every request, like `*` or `/v1/*`.

Reply:
If we miss a functionality in a core tool we should see how we can add it for us to have better tools. Regardless, I think you can just rely on the init hook for gaining the same effect for now and still enjoy the order, structure, consistency, validation, dependency injection, and other benefits we get from the FW.

Reply:
Implemented